internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -584,72 +600,39 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 	projectID := ids["project_id"]
 	clusterName := ids["cluster_name"]
 
-	var clusterResp *admin.ClusterDescription20240805
-
 	var replicationSpecs []map[string]any
-	if isUsingOldAPISchemaStructure(d) {
-		clusterOldSDK, resp, err := connV220240530.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-		if err != nil {
-			if resp != nil && resp.StatusCode == http.StatusNotFound {
-				d.SetId("")
-				return nil
-			}
-			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
-		}
 
-		if err := d.Set("disk_size_gb", clusterOldSDK.GetDiskSizeGB()); err != nil {
-			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "disk_size_gb", clusterName, err))
-		}
-		cluster, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-		if err != nil {
-			if resp != nil && resp.StatusCode == http.StatusNotFound {
-				d.SetId("")
-				return nil
-			}
-			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+	cluster, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
 		}
+		return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+	}
 
-		zoneNameToZoneIDs, err := getZoneIDsFromNewAPI(cluster)
-		if err != nil {
+	zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, projectID, clusterName, connV220240530.ClustersApi)
```

Comment:
[nit] Rename to oldShardConfigMeta?

Reply:
while more verbose I do like the current name as it give insight into how the resulting map is composed (zoneName -> replication spec metadata)